src/modules/account/components/ChangePasswordForm/useChangePasswordForm.ts

```diff
@@ -15,8 +14,9 @@ import { useActionField } from '@modules/account/contexts/ActionFieldContext';
 
 export const useChangePasswordForm = () => {
   const { t } = useTranslation();
-  const actionField = useActionField();
   const { user } = useAuth();
+  const [submitError, setSubmitError] = useState('');
+  const actionField = useActionField();
```

Comment:
Instead of this, which makes this component depends on ActionField, you can make an `onSubmit` inside ActionField and call the submit handler there instead.

Reply:
Same thing with the `LoginForm` and `RegisterForm` where we made them depend on the Modal instead. I think the general idea should be that any form component has `onSubmitSuccess` and `onSubmitError` prop.

Having `onSubmit` on the ActionField implies that ActionField is submitable which is not an expected behavior of the ActionField

Reply:
Oh wait, this is inside ChangePasswordForm, for some reason I thought this is a general component TextField. Reviewing code just after waking up be like.